### PR TITLE
:seedling: Add make unit-cover like in BMO

### LINKED ISSUE #440: Add unit-cover make target
This issue is reserved for the KubeCon EU 2024 contrib-fest. Please do not work on it before 2024-03-20.

Add a make target for running the unit tests and checking the test coverage. There is a target like this in [BMO](https://github.com/metal3-io/baremetal-operator/blob/b5c4120dd294e94cd63fede91321f38623293ec7/Makefile#L101-L108) that you can use for inspiration.
The goal is to be able to run `make unit-cover`. This should run all unit tests and print test coverage results.

### PR DESCRIPTION
**Pull request by lekaf974**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #440

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -33,3 +33,4 @@ releasenotes
 *~
 *.tmp
 .DS_Store
+.idea/
```
Comment (tuminoid) [inline .gitignore:36]: Line feed missing.

**Review by tuminoid — COMMENTED** [review]
/ok-to-test

File: .gitignore (hunk 1 above)
Comment (lekaf974) [reply .gitignore]: Removed the row from the file since it was more related to my setup I configured it on my side instead

**Review by lentzi90 — COMMENTED** [review]
/approve
Thanks!

**Comment by lekaf974** [comment]
Took a look on the test that failed and seems not to be related to those changes

**Review by tuminoid — COMMENTED** [review]
/lgtm

We can override the ubuntu integration test if it fails again.
Thanks for the contribution!

**Pull request merged**